contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -205,19 +205,21 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		h.htlc.RHash, h.htlc.Amt, h.htlcExpiry, currentHeight,
 		circuitKey, hodlChan, payload,
 	)
-	switch err {
-	case channeldb.ErrInvoiceNotFound:
-	case nil:
-		defer h.Registry.HodlUnsubscribeAll(hodlChan)
-
-		// Resolve the htlc directly if possible.
-		if event != nil {
-			return processHodlEvent(*event)
-		}
-	default:
+	if err != nil {
 		return nil, err
 	}
 
+	defer h.Registry.HodlUnsubscribeAll(hodlChan)
+
+	// If the event is non-nil, and the invoice is known to the registry
+	// (indicating that the htlc is not a forward), try to resolve the htlc
+	// directly.
+	if event != nil &&
+		event.ResolutionResult != invoices.ResultInvoiceNotFound {
```

Comment:
I would move the InvoiceNotFound refactor to a separate commit after introducing the `ResolutionResult`